app/jobs/mosaic_job.rb

```diff
@@ -1,16 +1,38 @@
 # frozen_string_literal: true
 class MosaicJob < ApplicationJob
+  class JobRunning < StandardError; end
   discard_on TileMetadataService::Error
+  retry_on JobRunning
   queue_as :low
   delegate :query_service, to: :metadata_adapter
 
-  def perform(resource_id)
-    resource = query_service.find_by(id: Valkyrie::ID.new(resource_id))
-    TileMetadataService.new(resource: resource).path
+  attr_reader :resource_id, :fingerprint
+  def perform(resource_id:, fingerprint:)
+    @resource_id = resource_id
+    @fingerprint = fingerprint
+    return unless valid_fingerprint?
+    raise JobRunning if currently_running?
+    TileMetadataService.new(resource: resource, generate: true).path
   end
 
   private
 
+    def valid_fingerprint?
```

Comment:
@hackartisan recommends adding some documentation about the race conditions we're avoiding here.

Reply:
Comments added.